### DISCUSSION TITLE
runtimeError: cusolver error: CUSOLVER_STATUS_EXECUTION_FAILED  in RandomResizedCrop

### DISCUSSION THREAD

**Discussion by shaibagon** [kornia/kornia#1691] [Potential Issue] [answered]
### Describe the bug

Calling `RandomResizedCrop` with reasonable parameters:
```
RandomResizedCrop(size=(args.size, args.size), scale=(0.8, 1.25), ratio=(0.95, 1.05))
```

results with this runtime error:
```
  File ".../kornia/augmentation/base.py", line 245, in forward
    output = self.apply_func(in_tensor, in_transform, self._params, return_transform)
  File ".../kornia/augmentation/base.py", line 203, in apply_func
    trans_matrix = self.compute_transformation(in_tensor, params)
  File ".../kornia/augmentation/augmentation.py", line 1310, in compute_transformation
    transform: torch.Tensor = get_perspective_transform(params['src'].to(input), params['dst'].to(input))
  File ".../kornia/geometry/transform/imgwarp.py", line 261, in get_perspective_transform
    X, _ = _torch_solve_cast(b, A)
  File ".../kornia/utils/helpers.py", line 101, in _torch_solve_cast
    out = solve(A.to(dtype), input.to(dtype))
RuntimeError: cusolver error: CUSOLVER_STATUS_EXECUTION_FAILED, when calling `cusolverDnSgetrf( handle, m, n, dA, ldda, static_cast<float*>(dataPtr.get()), ipiv, info)`

```

### Reproduction steps

```bash
Reproducing the error is a bit tricky, it does not happen 100% of the time, but rather occasionally.
I am using this augmentation of batches of 4 images of size ~256x256 pix.
Parameters are not extreme, I am not expecting any degeneracies in the transformation matrix. 
Moreover, this code ran smoothly many times in the past. The error started to appear when I upgraded my pytorch/cudatoolkit versions (see environment for details)
```

### Expected behavior

Expecting code not to crash when augmenting an image.

Currently crushing with:
```
RuntimeError: cusolver error: CUSOLVER_STATUS_EXECUTION_FAILED, when calling `cusolverDnSgetrf( handle, m, n, dA, ldda, static_cast<float*>(dataPtr.get()), ipiv, info)`
```

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
- PyTorch Version: 1.10.2, build py3.9_cuda11.3_cudnn8.2.0_0
- OS: Linux
- How you installed PyTorch: conda install
- Kornia version: 0.6.1, build pyhd8ed1ab_0
- Python version: 3.9.10
- CUDA/cuDNN version: cuda 11.3.1, cudatoolkit 11.3.1
- GPU: V100 
- Any other relevant information:
```


### Additional context

_No response_

**Comment by edgarriba** [comment]
@shaibagon thanks for reporting. Could you give us what's your input and the args structure ? I quickly tried here and all seems fine.

https://colab.research.google.com/drive/1sC6wXPNL_wkI4bqa5PfDSqmJA3864q8n?usp=sharing

**Comment by JanSellner** [comment, marked as answer]
I had the same issue with PyTorch 1.11 and CUDA 11.3 on an A100 GPU. I upgraded to CUDA 11.5 and could solve the issue (i.e. `pip install torch==1.11.0+cu115 --extra-index-url https://download.pytorch.org/whl/cu115`). In the [PyTorch Forums](https://discuss.pytorch.org/t/torch-c-linalgerror-after-calling-torch-linalg-inv-from-pytorch-1-11-using-cuda-11-3/148000), there is also a reproducible for a similar issue.
